### PR TITLE
[fga] extract WorkspaceService.start

### PR DESCRIPTION
**Pull request by akosyakov**
## Description
Extracts api.startWorkspace into WorkspaceService.startWorkspace, and adds some basic tests.

### <samp>🤖 Generated by Copilot at 7a1394b</samp>

This pull request refactors the server-side code for starting and stopping workspaces, by using a new `WorkspaceService` interface that consolidates the logic and checks the permissions, entitlements, limits, and other parameters. It also updates the tests and other components that depend on the old `WorkspaceStarter` interface, and improves the region guessing logic.

</details>

## Related Issue(s)

Fixes EXP-321

## How to test
 - start a regular workspace :heavy_check_mark: 

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ak-fga-wor2ec2aa2e5d</li>
	<li><b>🔗 URL</b> - <a href="https://ak-fga-wor2ec2aa2e5d.preview.gitpod-dev.com/workspaces" target="_blank">ak-fga-wor2ec2aa2e5d.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ak-fga_workspace_start-gha.15283</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 1]
```diff
@@ -944,38 +938,15 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             };
         }
 
-        if (!!workspace.softDeleted) {
```
Comment (akosyakov) [inline components/server/src/workspace/gitpod-server-impl.ts:947]: workspaceServer.getWorkspace does not check softDeleted, so probably it should stay?

Comment (akosyakov) [reply components/server/src/workspace/gitpod-server-impl.ts:947]: added softDeleted check in the workspace service

**Comment by geropl** [comment]
@akosyakov I rebased, changed 2 minor things and tested in the preview env. Is there anything specific you wanted to test?

**Comment by akosyakov** [comment]
@geropl not really, mostly workspace create/start and admin operations which we changed for blocking

**Comment by geropl** [comment]
Will test the blocking, one sec...

**Comment by geropl** [comment]
Works. :heavy_check_mark:

**Comment by akosyakov** [comment]
landing?

**Review by geropl — APPROVED** [review]
:heavy_check_mark:

**Pull request merged**